test/python.gdextension

```diff
@@ -5,7 +5,7 @@ entry_symbol = "python_extension_init"
 
 [libraries]
 
-macos = "res://bin/macos/libgodot-python.macos.framework"
+macos = "res://bin/macos/libgodot-python.macos.universal.dylib"
```

Comment:
I don't know the difference here, but maybe there's a tag to be added after `macos` so that both can be listed?

Reply:
The difference between the dylib is explained in my PR text. TL;DR dylib and framework use the same binary but with a slightly different folder structure. Both cannot be supported since the gdextension must choose one path for the tags at hand.